javascript/ql/lib/semmle/javascript/frameworks/SQL.qll

```diff
@@ -604,6 +662,18 @@ private module Spanner {
       this = transaction().getMember(["run", "runStream", "runUpdate"]).getACall()
       or
       this = batchTransaction().getMember("createQueryPartitions").getACall()
+      or
+      this = snapshot().getMember(["run", "runStream"]).getACall()
+    }
+
+    override DataFlow::Node getAResult() {
+      PromiseFlow::loadStep(this.getALocalUse(), result, Promises::valueProp())
+      or
+      (this = database().getMember("run").getACall() and result = this.getCallback(_).getParameter(1))
+      or
+      (this = transaction().getMember("run").getACall() and result = this.getCallback(_).getParameter(1))
+      or
+      (this = snapshot().getMember("run").getACall() and result = this.getCallback(_).getParameter(1))
```

Comment:
I think you should be able to simplify this using set-literals to
```ql
this = [database(), transaction(), snapshot()].getMember("run").getACall() and result = this.getCallback(_).getParameter(1))
```

Reply:
clever, thx! :)